tests/integration/pilot/cni/cni_test.go

```diff
@@ -57,32 +57,34 @@ func TestCNIReachability(t *testing.T) {
 			if err != nil {
 				ctx.Fatal(err)
 			}
-			rctx := reachability.CreateContext(ctx, false)
+			rctx, err := reachability.CreateContext(ctx, false)
```

Comment:
Can we move the CreateContext into the test main?

Reply:
Also, we have a somewhat similar setup under `tests/integration/pilot/common` for reachability/traffic tests. Could that be merged with the security tests?

Reply:
> Can we move the CreateContext into the test main?
Yeah we can but currently the following four tests which use this function has different control plan configuration, thus it looks like we can not directly reuse the echo instances among them. 
tests/integration/security/mtls_first_party_jwt/strict_test.go
tests/integration/security/mtlsk8sca/strict_test.go
tests/integration/security/reachability_test.go
tests/integration/pilot/cni/cni_test.go
Currently the echo instances are reused within the https://github.com/istio/istio/blob/master/tests/integration/security/util/reachability/context.go#L125.

Reply:
> Also, we have a somewhat similar setup under `tests/integration/pilot/common` for reachability/traffic tests. Could that be merged with the security tests?

Yeah possible. One concern I have will be the focuses of different subfolders. `security` will be focused on different security related scenario set up. Not sure if it is proper to move all reachability related tests into `security`. Maybe we can move the reachability related setup into `pkg\test\framework` later if `pilot` and `security` want to share some common setup and checks. @myidpt what do you think about this?

Reply:
If the control-plane config is per-suite, the echos should be able to be per-suite. That way, if multiple TestXXXX methods exist in this package later, they can re-use the same workloads. It's just a best practice, not something that has immediate gains.